### DISCUSSION TITLE
Allow user to change email with new OTP feature

### DISCUSSION THREAD

**Discussion by jjones3535a** [pocketbase/pocketbase#5765] [Q&A]
I'm using v23 and have OTP working.  I use a combination of requestOTP and authWithOTP to create and login a user.  I'd like to allow the user typical profile management stuff like updating name, avatar and email.  I know there is requestEmailChange(newEmail) method, but that sends an email and requires the user to enter a password to update their account's email.  However, with OTP I've created a user but set an hashed password that they don't have to manage so they can just login with a "magic link".  

How would I allow a user to update their email address if I'm not using passwords for authentication?

**Comment by ganigeorgiev** [comment]
I'm not really sure but the problem is the same as in another older similar issue with OAuth2 that it is currently "on hold".

For the OAuth2 when creating a new user we generate a random password and there was suggestion to return that random password in the first auth response so that users can treat it as "security code" but I'm still not sure if this is a good approach and it remains in my todo for later consideration.

The easiest workaround is to send a password reset email to your users so that they can set their own explicit password and then followup with the email change confirmation but obviously this is not very user friendly.

How do you imagine the ideal flow for confirming the email change in your situation?

**Reply by jjones3535a** [reply]
I think a flow for confirming an email change in the case where the user does not have a password would be like this:
1) User is logged in an on a Profile page, which lists their current email and has an input to enter a new email with a submit button
2) User enters new email, and PB sends a confirmEmailChange email to the email address the user just entered
3) User goes to email and clicks link and the link automatically looks up user from user collection and updates that record with the new email address

**Reply by ganigeorgiev** [reply]
No, I don't think changing the email with just the confirmation link from the new email address is OK. It is a critical operation and we need an extra confirmation in addition to the current logged in state (e.g. the password or sending a temp "security confirmation code" to the original email).

I'll think a little more on it together with the OAuth2 case sometime in the future, but for now remains a low priority.
